src/main/java/seedu/task/logic/commands/EditTaskCommand.java

```diff
@@ -0,0 +1,100 @@
+package seedu.task.logic.commands;
+
+import seedu.task.commons.exceptions.IllegalValueException;
+import seedu.task.model.item.Description;
+import seedu.task.model.item.Name;
+import seedu.task.model.item.ReadOnlyTask;
+import seedu.task.model.item.Task;
+import seedu.task.model.item.UniqueTaskList;
+import seedu.task.model.item.UniqueTaskList.DuplicateTaskException;
+import seedu.taskcommons.core.Messages;
+import seedu.taskcommons.core.UnmodifiableObservableList;
+
+/**
+ * Executes editing of tasks according to the input argument.
+ * @author kian ming
+ */
+public class EditTaskCommand extends EditCommand  {
+
+    public static final String MESSAGE_EDIT_TASK_SUCCESS = "Edited Task: %1$s";
+    private static final Boolean TASK_DEFAULT_STATUS = false;
+    public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in the task book";
+    
+    private Name newName;
+    private Description newDescription;
+//    private Deadline newDeadline;
+    private boolean isNameToBeEdit;
+    private boolean isDescriptionToBeEdit;
+//    private boolean isDeadlineToBeEdit;
+    
+    /**
+     * Convenience constructor using raw values.
+     * Only fields to be edited will have values parsed in.
+     * TODO: 
+     *  1. allow edit tasks with deadline
+     *  
+     * @throws IllegalValueException
+     *             if any of the raw values are invalid
+     */
+    public EditTaskCommand(Integer index,
+                           boolean isNameToBeEdited, String name, 
+                           boolean isDescriptionToBeEdited, String description
+//                           boolean isDeadlineToBeEdited, String deadline
+                           ) throws IllegalValueException {
+        
+        setTargetIndex(index);
+        this.isNameToBeEdit = isNameToBeEdited;
+        this.isDescriptionToBeEdit = isDescriptionToBeEdited;
+        
+        newName = null;
```

Comment:
Yeah, it is probably redundant. The objects will automatically be intitialised to null too if no values are assigned. But I guess it's minor so can just leave it here.